Source/WebCore/platform/graphics/gstreamer/eme/WebKitOpenCDMDecryptorGStreamer.cpp

```diff
@@ -192,7 +193,13 @@ static bool webKitMediaOpenCDMDecryptorDecrypt(WebKitMediaCommonEncryptionDecryp
         return false;
     }
 
-    WebKitOpenCDMDecryptPrivate* priv = GST_WEBKIT_OPENCDM_DECRYPT_GET_PRIVATE(WEBKIT_OPENCDM_DECRYPT(self));
+    GstMapInfo keyIDBufferMap;
+    if (!gst_buffer_map(keyIDBuffer, &keyIDBufferMap, GST_MAP_READ)) {
+        gst_buffer_unmap(ivBuffer, &ivMap);
+        gst_buffer_unmap(buffer, &map);
```

Comment:
I have discussed at some point with Phil that this kind of code is ugly and it would be awesome to create something like GstSmartMap that could handle this code gracefully. Because even if we are careful, we are (still) humans and we are going to mess up at some point.

This could be also good for upstream and bring it down.